agent/bench-scripts/postprocess/trafficgen-postprocess

```diff
@@ -174,50 +212,55 @@ while ( <TXT> ) {
 }
 close TXT;
 my $json_result = from_json($json_text);
+$json_text = "";  # Release this huge string to relieve my IDE's debugger
 my %last_trial = %{ $$json_result{'trials'}[-1] };
 
-#print Dumper $json_result;
-#exit;
-
 foreach my $trial_param ( keys %{ $last_trial{'trial_params'} } ) {
 	if ( $trial_param eq 'output_dir' ) {
 		# The output_dir parameter is different per sample; we explicitly omit it
 		# so that when process-iteration-samples runs it will properly merge the
 		# same sample data sets together.
 		next;
 	}
-	if ( $last_trial{'trial_params'}{$trial_param} ) {
+	if (defined($last_trial{'trial_params'}{$trial_param})) {
 		$config{$trial_param} = $last_trial{'trial_params'}{$trial_param}
 	}
```

Comment:
This change is causing `test-22` to fail:  the previous code would omit any parameters with zero or empty-string values, whereas now they are included in the result.  This pulls in `"max_loss_pct"` and `"min_rate"`, which have zero values, which is probably appropriate (ditto `"trial_gap"`); however, it also pulls in `"pre_trial_cmd"`, which is an empty string, which might not be appropriate.  (The inclusion of `"max_loss_pct"` has the knock-on effect of causing the `"uid"` value to change.)

Reply:
Is this a bug or a feature?  (I.e., should I be reverting this change or updating the `test-22` gold file?)

Reply:
Sounds like a bug to me, but I am wondering if empty strings might be worth excluding.

Reply:
So, @portante, this issue is now the sole blocker for this PR (at least, until the next round of code review...).  Should I

- revert this change,
- accept this change and update the gold file, or
- split the difference:  report zeros but omit empty strings?

Reply:
Filed issue #2462 to track.

Reply:
It turned out that reverting this change didn't resolve the problem.  🙁

So, I went ahead and added code to exclude fields with empty string values.  Unfortunately, there is considerably subtlety in Perl's type-less scalars, and the JSON formatter that we use disambiguates them based on how they were _last **used**_, so checking to see that a (numeric) value is a not an empty _string_ caused several of the numeric fields to then be dumped **as strings**.

So, I had to add logic to detect numeric values based on their contents, and, for things which looked numeric, to re-numerify them.  Unfortunately, JSON boolean values appear to be numeric, but we want them formatted as neither numbers nor strings, so I had to add further logic to ensure that they come through unscathed.

Nevertheless, after much wrangling, I am satisfied with the result.  It does add a few fields to the output, and the UID does change, but I think the latter is an intended result, and the former are not bad things.  (And, it aligns well with the `b0.69` results, although I'm not sure what we did in 0.71 which requires extra logic.)